libs/service-portal/documents/src/components/DocumentRenderer/DocumentRendererV2.tsx

```diff
@@ -0,0 +1,73 @@
+import { HtmlDocument } from './HTMLDocument'
+import { PdfDocWithModal } from './PdfDocument'
+import { UrlDocument } from './UrlDocment'
+import NoPDF from '../NoPDF/NoPDF'
+import { messages } from '../../utils/messages'
+import { ActiveDocumentType2 } from '../../lib/types'
+import { useLocale } from '@island.is/localization'
+import { customUrl } from '../../utils/customUrlHandler'
+
+const parseDocumentType = (document: ActiveDocumentType2) => {
+  const doc = document.document
+  const overviewUrl = document.downloadUrl
+
+  if (doc?.type === 'HTML') {
+    return 'html'
+  }
+  if (doc?.type === 'PDF') {
+    return 'pdf'
+  }
+  if (doc?.type === 'URL' || overviewUrl) {
+    return 'url'
+  }
+  return 'unknown'
+}
+
+type DocumentRendererProps = {
+  document: ActiveDocumentType2
+}
+
+export const DocumentRenderer: React.FC<DocumentRendererProps> = ({
+  document,
+}) => {
+  const { formatMessage } = useLocale()
+  const type = parseDocumentType(document)
+
+  if (type === 'unknown') return <NoPDF text={formatMessage(messages.error)} />
+
+  if (type === 'html') {
+    return <HtmlDocument html={document.document?.value ?? ''} />
+  }
+
+  if (type === 'pdf') {
```

Comment:
Why all this extra stuff when you can just ` if (document.document.type === 'PDF')` or something